k9mail/src/main/java/com/fsck/k9/mailstore/MessageInfoExtractor.java

```diff
@@ -25,6 +26,11 @@ public MessageInfoExtractor(Context context, Message message) {
     }
 
     public String getMessageTextPreview() throws MessagingException {
+        if (CryptoHelper.isPgpInlineEncrypted(message)
+                || CryptoHelper.isPgpMimeEncrypted(message)
+                || CryptoHelper.isSMimeEncrypted(message)) {
+            return "*Encrypted*";
```

Comment:
Of course. Not to mention that as it stands, the implementation is more than a little magic number-ish. I'll try to get this fixed tonight.

Thanks artbristol.